src/cmd/linuxkit/pkglib/build.go

```diff
@@ -191,3 +201,62 @@ func (p Pkg) Build(bos ...BuildOpt) error {
 
 	return nil
 }
+
+type buildCtx struct {
+	sources []string
+}
+
+// Copy iterates over the sources, tars up the content after rewriting the paths.
+// It assumes that sources is sane, ie is well formed and the first part is an absolute path
+// and that it exists. NewFromCLI() ensures that.
+func (c *buildCtx) Copy(w io.WriteCloser) {
+	tw := tar.NewWriter(w)
+	defer func() {
+		tw.Close()
+		w.Close()
+	}()
+
+	for _, source := range c.sources {
+		tmp := strings.Split(source, ":")
+		src := tmp[0]
+		dst := tmp[1]
+		log.Debugf("Adding to build context: %s -> %s", src, dst)
+
+		f := func(p string, i os.FileInfo, err error) error {
+			if err != nil {
+				log.Warnf("ctx: Walk error on %s: %v", p, err)
+				return nil
```

Comment:
This function appears to warn and eat all the errors -- should it not return the error and fail the build?

Reply:
There does not seem to be a way to return an error all the way back to the pipe and no way yo handle the error. I could change this to log.Fatal()` or similar to exit

Reply:
Returning an error in this function propagates to the error return of the `filepath.Walk`, you just need to handle it there instead of ignoring it. I think.

Maybe you need to propagate it further, like back the `if dr.ctx != nil {` block goroutine, an `errgroup` there would save you messing around with channels manually.